src/components/SubStepForms/FullNameStep.tsx

```diff
@@ -64,15 +73,33 @@ function FullNameStep<TFormID extends keyof OnyxFormValuesMapping>({
             const errors = ValidationUtils.getFieldRequiredErrors(values, stepFields);
 
             const firstName = values[firstNameInputID as keyof FormOnyxValues<TFormID>] as string;
-            if (firstName && !ValidationUtils.isValidLegalName(firstName)) {
+            if (!ValidationUtils.isRequiredFulfilled(firstName)) {
                 // @ts-expect-error type mismatch to be fixed
                 errors[firstNameInputID] = translate('common.error.fieldRequired');
+            } else if (!ValidationUtils.isValidLegalName(firstName)) {
+                // @ts-expect-error type mismatch to be fixed
+                errors[firstNameInputID] = translate('privatePersonalDetails.error.hasInvalidCharacter');
+            } else if (firstName.length > CONST.LEGAL_NAME.MAX_LENGTH) {
+                // @ts-expect-error type mismatch to be fixed
+                errors[firstNameInputID] = translate('common.error.characterLimitExceedCounter', {
+                    length: firstName.length,
+                    limit: CONST.LEGAL_NAME.MAX_LENGTH,
+                });
             }
 
             const lastName = values[lastNameInputID as keyof FormOnyxValues<TFormID>] as string;
-            if (lastName && !ValidationUtils.isValidLegalName(lastName)) {
+            if (!ValidationUtils.isRequiredFulfilled(lastName)) {
```

Comment:
so now we have much more specific validation as a default - does it apply to all the places we use this common component?

Reply:
I think Yes. As we used `FullNameStep` mostly for validating legal name, I think we should also use legal name validation as the default validation, what do you think?

Reply:
I'm not saying it's wrong, but previously we checked only for required constraint in other places - I think we should test it at least in these places

Reply:
Ah yeah. Agreed.